internal/pkg/util/fs/overlay/overlay_item_linux.go

```diff
@@ -155,6 +155,18 @@ func (i *Item) Mount() error {
 	return nil
 }
 
+// GetMountDir returns the path to the directory that will actually be mounted
+// for this overlay. For squashfs overlays, this is equivalent to the
+// Item.StagingDir field. But for all other overlays, it is the "upper"
+// subdirectory of Item.StagingDir.
+func (i Item) GetMountDir() string {
```

Comment:
This still isn't quite right. If I am overlaying a directory read-only then it doesn't have to have an upper. If it doesn't have an upper, then the directory content should be overlayed directly:

```
$ ls overlaydirro
bob

$ singularity exec -u --overlay overlaydirro:ro docker://alpine ls /
INFO:    Using cached SIF image
INFO:    Converting SIF file to temporary sandbox...
bin          environment  lib          opt          run          srv          usr
bob          etc          media        proc         sbin         sys          var
dev          home         mnt          root         singularity  tmp
INFO:    Cleaning up image...

$ singularity exec --oci --overlay overlaydirro:ro docker://alpine ls /
2023/06/08 08:59:06  info unpack layer: sha256:8a49fdb3b6a5ff2bd8ec6a86c05b2922a0f7454579ecc07637e94dfd1d0639b6
FATAL:   failed to mount /tmp/oci-bundle3793341770/rootfs: no such file or directory
```

The logic missing here from the native engine is...

https://github.com/sylabs/singularity/blob/60a6afb2184a8684ea1c11cfcf73650d85294e74/internal/pkg/runtime/engine/singularity/container_linux.go#L1050